### PR TITLE
fix-issue-282-273

### PR DESCRIPTION
**Pull request by qiaozha**

### CODE REVIEW

File: src/plugins/azgenerator/TemplateAzureCliParams.ts [hunk 1]
```diff
@@ -171,8 +171,38 @@ function getCommandBody(model: CodeModelAz, needUpdate: boolean = false, needGen
 
                         argument = "        c.argument('" + parameterName + "'";
                         argument += ", options_list=['--" + parameterName.substr(0, parameterName.length - 1) + "']";
+                    } else if (parameterName.endsWith('name') && parameterName.replace(/_name|_/g, '') == model.CommandGroup_DefaultName.toLowerCase()) {
```
Comment (zikalino) [inline src/plugins/azgenerator/TemplateAzureCliParams.ts:174]: shouldn't be endsWith('_name')?

Comment (qiaozha) [reply src/plugins/azgenerator/TemplateAzureCliParams.ts:174]: no. this is to fix issue 282. to automatically add options_list in main resource name

Comment (qiaozha) [reply src/plugins/azgenerator/TemplateAzureCliParams.ts:174]: I use the comamnd group name and parameter name to match whether this parameter is main resource name or not.

File: src/plugins/azgenerator/CodeModelAz.ts [hunk 2]
```diff
@@ -71,6 +71,7 @@ export interface CodeModelAz
     CommandGroup: OperationGroup;
     CommandGroup_Name: string;
     CommandGroup_Help: string;
+    CommandGroup_DefaultName: string;
```
Comment (zikalino) [inline src/plugins/azgenerator/CodeModelAz.ts:74]: where is it set?

Comment (qiaozha) [reply src/plugins/azgenerator/CodeModelAz.ts:74]: it already set in here https://github.com/Azure/autorest.az/blob/master/src/plugins/azgenerator/CodeModelAzImpl.ts#L467 Just only in CodeModelAzImpl.ts not in  CodeModelAz.ts

Comment (zikalino) [reply src/plugins/azgenerator/CodeModelAz.ts:74]: oh, ok, so it was implemented, just not exposed

File: src/plugins/azgenerator/TemplateAzureCliParams.ts (hunk 1 above)
Comment (zikalino) [reply src/plugins/azgenerator/TemplateAzureCliParams.ts:174]: ok

**Pull request merged**